### PR TITLE
make confirms actually work + make them work without an offset manager

### PR DESCRIPTION
**Pull request by ryedin**
- create a confirms channel per message so that tracking is accurate
- enable configuring pipelines to await for confirms without needing an offset manager
- make the rethinkdb writer bulk insert operations properly track confirmations on all messages
- make the pipeline generically and accurately await message confirmations before passing to the next stage
- fixed bugs in existing transformers where message confirmation channels would get dropped

### CODE REVIEW

File: cmd/transporter/goja_builder.go [hunk 1]
```diff
@@ -82,10 +82,11 @@ type Transporter struct {
 }
 
 type config struct {
-	LogDir             string `json:"log_dir"`
+	LogDir             string `json:"xlog_dir"`
```
Comment (yasserf) [inline cmd/transporter/goja_builder.go]: this meant to xlog now?

Comment (ryedin) [reply cmd/transporter/goja_builder.go]: yeah this was a bug in their code. If you checkout their readme they specify it as `xlog_dir` and there is even a commit that mentions that it was changed from `log_dir`, but they missed this location, so the readme was essentially incorrect

Comment (ryedin) [reply cmd/transporter/goja_builder.go]: just a tangential fix... the first thing I tried was turning the commit log on. when it didn't work, I hunted this down

Comment (yasserf) [reply cmd/transporter/goja_builder.go]: 👍 good catch!

**Pull request merged**